### ISSUE TITLE
Thrustmaster GPX and GP XID PRO controllers not detected in steam linux anymore.

### ISSUE THREAD

**Issue by byoso** [ValveSoftware/steam-for-linux#5621]
#### Your system information

* Steam client version: Aug 8 2018
* Distribution:  Linux Mint 18.3
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
Thrustmaster gamepads (GPX and GP XID PRO)are not detected anymore  in steam since a recent update. The controller is not detected in the menu steam/settings/controller/general controller settings The standard xbox360 gamepad (by microsoft) still works fine in steam.
The thrustmaster gamepads work fine out of steam, tested with jstest-gtk, or in none-steam games, also tested on the windows steam version (same computer dual booting windows), detected and works fine.

#### Steps for reproducing this issue:

1. plug a Thrustmaster gpx (or gp xid pro) gamepad 
2. launch steam
3. open the menu steam/settings/controllers/controllers settings, the controller is not detected in steam, but it works out of it.

I hope it could help, thanks for reading anyway.

**Comment by doctortee** [comment]
Same issue here (using Thrustmaster GP XID). It is detected by the system, and some games, but it isn't detected properly by steam.

Note that steam sometimes display a message about the controller in logs
```

Local Device Found
  type: 044f b326
  path: sdl://0
  serial_number:  - 0
  Manufacturer: 
  Product:      Thrustmaster Gamepad GP XID
  Release:      402
  Interface:    -1
```

Using Steam beta on Debian Testing

**Comment by tomgar** [comment]
I confirm this on Archlinux. Wrote something about it on #5479 .

**Comment by tomgar** [comment]
Well, I discovered how to fix this issue... it turns out that Steam is quite dependent on SDL's game controller database. There's no entry for the Thrustmaster GPX inside that DB. I just added a pull request here to include it: 
https://github.com/gabomdq/SDL_GameControllerDB/pull/229

In the mean time you can run Steam like this to add support for your Thrusmaster GPX:
SDL_GAMECONTROLLERCONFIG='03000000c6240000025b000002020000,Thrustmaster GPX Gamepad,a:b0,b:b1,x:b2,y:b3,back:b6,guide:b8,start:b7,leftstick:b9,rightstick:b10,leftshoulder:b4,rightshoulder:b5,dpup:h0.1,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,leftx:a0,lefty:a1,rightx:a3,righty:a4,lefttrigger:a2,righttrigger:a5,platform:Linux,' steam

Export that variable (without calling steam at the end of course) to your .profile file in your $HOME or similar so it's executed at every reboot, then you'll be able to run Steam from your desktop and get such controller support automatically.

I have a Thrustmaster GPX, not a GP XID PRO, so if this fixes the issue for you then, you need to make a similar line for it. It's really easy with the application you can download from: http://generalarcade.com/gamepadtool/

**Comment by Wenzel** [comment]
Hi @tomgar ,
I tried your solution today but it doesn't work for me.
I'm running KDE, and the desktop detects my controller (`Thrustmaster T Mini Wireless (/dev/input/js0)`)

But Steam doesn't.
Also, i don't have the same model as you, so maybe I just have to change `SDL_GAMECONTROLLERCONFIG` ? How did you determine this value ??

btw, when i run Steam, it says that i detects the controller in the log, but still can't use it:
~~~
Local Device Found
  type: 044f d007
  path: sdl://0
  serial_number:  - 0
  Manufacturer: 
  Product:      Thrustmaster T Mini Wireless
  Release:      100
  Interface:    -1
~~~

Thanks !

**Comment by tomgar** [comment]
> Hi @tomgar ,
> I tried your solution today but it doesn't work for me.
> I'm running KDE, and the desktop detects my controller (`Thrustmaster T Mini Wireless (/dev/input/js0)`)
> 
> But Steam doesn't.
> Also, i don't have the same model as you, so maybe I just have to change `SDL_GAMECONTROLLERCONFIG` ? How did you determine this value ??
> 
> btw, when i run Steam, it says that i detects the controller in the log, but still can't use it:
> 
> ```
> Local Device Found
>   type: 044f d007
>   path: sdl://0
>   serial_number:  - 0
>   Manufacturer: 
>   Product:      Thrustmaster T Mini Wireless
>   Release:      100
>   Interface:    -1
> ```
> 
> Thanks !

Of course, @Wenzel , the environment variable needs to be specially made for your controller. The first big number of that variable is the GUID which is specific per controller, because each controller is unique and needs a specific configuration. I explained in the last paragraph of my comment how to produce such environmental variable. Use that program and you'll have it.

**Comment by Wenzel** [comment]
I missed the tool you mentionned, sorry.

I was able to generate the right configuration for my controller and steam now recognizes it :+1: 
Let's play Hollow Knight :)

Many thanks !

**Comment by kisak-valve** [comment]
Hello @rcgordon, @slouken, is @tomgar's SDL_GAMECONTROLLERCONFIG something you'd like to pull into upstream SDL or would it be more useful for @tomgar and @Wenzel to file an [upstream bug report](https://bugzilla.libsdl.org/buglist.cgi?component=joystick&product=SDL&resolution=---) to track this kind of info closer to the project?

**Comment by j-a-k-g** [comment]
I just got a GP XID PRO which is detected and working OOTB on current Arch and stable Steam (718904662). Guess this got fixed at some point.
